### PR TITLE
 Enable -Xsource:2.13 and resolve subsequent ambiguous implicits

### PR DESCRIPTION
**Pull request by TomasMikula**
Taking advantage of scala/scala#6069

### CODE REVIEW

File: base/shared/src/main/scala/scalaz/data/Forall.scala [hunk 1]
```diff
@@ -48,26 +48,13 @@ trait ForallModule {
   object Unapply {
     implicit def unapply[G[_]]: Unapply[∀[G]] { type F[A] = G[A] } =
```
Comment (jdegoes) [inline base/shared/src/main/scala/scalaz/data/Forall.scala]: Does this subsume all the deleted cases?

File: base/shared/src/main/scala/scalaz/data/Forall.scala [hunk 2]
```diff
@@ -48,26 +48,13 @@ trait ForallModule {
...
-
-    implicit def unapply3[P[_, _], G[_], H[_]]: Unapply[∀[λ[α => P[G[α], H[α]]]]] { type F[A] = P[G[A], H[A]] } =
-      new Unapply[∀[λ[α => P[G[α], H[α]]]]] { type F[A] = P[G[A], H[A]] }
   }
 }
 
 trait ForallSyntax {
   import ForallSyntax._
 
   implicit def toForallOps[F[_]](a: ∀[F]): Ops[F] = new Ops[F](a)
```
Comment (jdegoes) [inline base/shared/src/main/scala/scalaz/data/Forall.scala]: Ditto.

File: base/shared/src/main/scala/scalaz/data/Forall2.scala [hunk 3]
```diff
@@ -35,26 +35,13 @@ trait Forall2Module {
   object Unapply {
     implicit def unapply[G[_, _]]: Unapply[∀∀[G]] { type F[A, B] = G[A, B] } =
       new Unapply[∀∀[G]] { type F[A, B] = G[A, B] }
```
Comment (jdegoes) [inline base/shared/src/main/scala/scalaz/data/Forall2.scala]: Same here.

File: base/shared/src/main/scala/scalaz/data/Forall.scala (hunk 1 above)
Comment (TomasMikula) [reply base/shared/src/main/scala/scalaz/data/Forall.scala]: Yes. We have [examples](https://github.com/scalaz/scalaz/blob/766c09ffc0f6ef243da94868fcd61a5c892d7a19/example/shared/src/main/scala/scalaz/example/ForallUsage.scala) that previously needed those other.

File: base/shared/src/main/scala/scalaz/data/Forall.scala (hunk 2 above)
Comment (TomasMikula) [reply base/shared/src/main/scala/scalaz/data/Forall.scala]: Likewise, plus scalac now contains some very similar [test cases](https://github.com/scala/scala/blob/c2a5883891a68180b143eb462c8b0cebc8d3b021/test/files/run/hk-typevar-unification.scala) for these. 😈

File: base/shared/src/main/scala/scalaz/data/Forall2.scala (hunk 3 above)
Comment (TomasMikula) [reply base/shared/src/main/scala/scalaz/data/Forall2.scala]: Yup.

File: base/shared/src/main/scala/scalaz/data/Forall.scala (hunk 2 above)
Comment (jdegoes) [reply base/shared/src/main/scala/scalaz/data/Forall.scala]: ❤️ ❤️ ❤️

**Pull request merged**